test/simplify_reshapes_test.cpp

```diff
@@ -328,4 +329,42 @@ TEST_CASE(concat_transpose3)
     EXPECT(migraphx::any_cast<migraphx::op::concat>(new_concat->get_operator()).axis == 1);
 }
 
+TEST_CASE(nested_concat1)
+{
+    migraphx::program p;
+    auto s       = migraphx::shape{migraphx::shape::float_type, {1, 2, 3, 4}};
+    auto x       = p.add_parameter("x", s);
+    auto y       = p.add_parameter("y", s);
+    auto concat1 = p.add_instruction(migraphx::op::concat{1}, x, y);
+    auto concat2 = p.add_instruction(migraphx::op::concat{1}, y, x);
+    auto concat3 = p.add_instruction(migraphx::op::concat{1}, concat1, concat2);
+    p.add_instruction(pass_op{}, concat3);
+    auto out_shape = p.get_shape();
+    auto n         = std::distance(p.begin(), p.end());
+    p.compile(simplify_reshapes_target{});
+    EXPECT(p.get_shape().lens() == out_shape.lens());
+    EXPECT(std::distance(p.begin(), p.end()) == n - 2);
+    EXPECT(std::count_if(p.begin(), p.end(), [](auto ins) { return ins.name() == "concat"; }) == 1);
+}
+
+TEST_CASE(nested_concat2)
```

Comment:
can you rename the test to something like "nested_concat_params_plus_literal"

Reply:
I renamed it to `nested_concat_partial`.